tests/Base.nut

```diff
@@ -0,0 +1,128 @@
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+// EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+// OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+// "Promise" symbol is injected dependency from ImpUnit_Promise module,
+// while class being tested can be accessed from global scope as "::Promise".
+
+@include "github:electricimp/ConnectionManager/ConnectionManager.lib.nut"
+
+const MESSAGE_NAME = "test";
+const MESSAGE_WITHOUT_RESPONSE = "test_no_response";
+const MESSAGE_WITH_DELAY = "test_with_delay";
+const MESSAGE_WITH_LONG_DELAY = "test_with_huge_delay";
+const MESSAGE_WITH_SHORT_DELAY = "test_with_small_delay";
+const MESSAGE_DESTRUCTIVE_RESEND = "test_destructive_resend";
+const MESSAGE_DESTRUCTIVE_RESEND_RESPONSE = "test_destructive_resend_response";
+
+const REPLY_NO_MESSAGES = "No messages";
+const BASIC_MESSAGE = "basic message";
+
+const MESSAGE_WITH_DELAY_SLEEP = 2;
+const MESSAGE_WITH_DELAY_LONG_SLEEP = 8;
+const MESSAGE_WITH_DELAY_SHORT_SLEEP = 1;
+
+const ERR_REQ_RES_NOT_IDENTICAL = "Request and response messages are not identical";
+const ERR_REQ_RES_IDENTICAL = "Request and response messages are identical";
+
+// These variables are used to correctly receive messages in the conditions of execution of the chain of tests.
+// Each test will contain these two parameters of MessageManager constructor.
+// MessageManager({
+//    "firstMessageId":  msgId,
+//    "nextIdGenerator": msgIdGenerator
+// })
+// This approach will be applied for all tests of this library.
+local msgId = 0;
+local msgIdGenerator = function() {
+    return ++msgId;
+}.bindenv(this);
+
+function isAgentSide() {
+    return imp.environment() == ENVIRONMENT_AGENT;
+}
+
+function infoAboutSide() {
+    info("Tests will be performed on the " + (isAgentSide() ? "agent" : "device") + "-side");
+}
+
+function assertDeepEqualWrap(expected, actual, message = null, compare = true) {
+    if (compare) {
+        local compareString = "Type: " + typeof actual + ". " +
+                              "Expected: '" + expected + "'. " +
+                              "Got: '" + actual + "'.";
+        if (message == null) {
+            message = compareString;
+        } else {
+            message = message + ". " + compareString;
+        }
+    }
+    if (typeof actual == "blob") {
+        assertDeepEqual(expected.tostring(), actual.tostring(), message);
+    } else {
+        assertDeepEqual(expected, actual, message);
+    }
+}
+
+function getConnectionManager(dummy = true, options = null) {
```

Comment:
Connection manager is a device side library. So it probably makes sense to add this check here.